Fix Search Bar Placeholder Text Overflow.

aditii2712 opened the pull request:
Fixes #1184

This PR fixes the issue where the placeholder text in the search bar was overflowing beyond the input field’s width. The fix ensures that the placeholder text stays within the search bar.

**Screenshot including the fix:-**


![Screenshot 2025-03-06 at 1 09 14 AM](https://github.com/user-attachments/assets/bef27ef1-4a52-4c50-88aa-8e327827493e)

aditii2712 commented:
@kelson42 Please take a look when get a chance.

Thanks!

kelson42 commented:
@aditii2712 Not a proper fix, it works here, but no garanty that it works with an other title.

aditii2712 commented:
@kelson42 Regret, I had missed that we're using the same header across the application. 

I've implemented another fix for the placeholder text overflow in the search box by:
1. Using text truncation for the over-flowing text with ellipsis (...)
2. Aloo, Adjusted the responsive width parameters for better display across device sizes

PTAL when you get a chance..

Adding a vid to show the updates. 


https://github.com/user-attachments/assets/4698712d-9726-47e9-9395-51ef3fe7be36

aditii2712 commented:
@kelson42 Please take a look when get a chance.

aditii2712 commented:
Hi @kelson42, Would appreciate your review. Thanks!

kelson42 reviewed (approved):
From a user perspective, looks good to me. @veloman-yunkan please review for the tech part.

veloman-yunkan reviewed (commented):
Don't these changes affect text input? What if I type long text in the searchbox?

kelson42 commented:
@veloman-yunkan I don't see any problem when typed text is long, it behaves in a normal manner.

veloman-yunkan commented on static/skin/taskbar.css:
```diff
@@ -32,6 +32,13 @@
 
 .kiwix_searchform {
     width: 20em;
+    max-width: 40%;
```
This results in too short search field on narrow screens.

veloman-yunkan commented on static/skin/taskbar.css:
```diff
@@ -170,7 +177,7 @@ a.suggest, a.suggest:visited, a.suggest:hover, a.suggest:active {
 
     .kiwix_searchform {
-        width: 36%;
+        width: 50%;
```
Even if possibly an improvement in responsiveness, it is rather a matter of taste and has nothing to do with the issue that this PR addresses. I am not aware of the reasons behind the current value of this setting but it looks like it wasn't just random. So I am in favour of preserving it.

veloman-yunkan commented on static/skin/taskbar.css:
```diff
@@ -180,10 +187,10 @@ a.suggest, a.suggest:visited, a.suggest:hover, a.suggest:active {
 @media(max-width:419px) {
     .kiwix_searchform {
-        width: 80%;
+        width: 85%;
```
Same here

veloman-yunkan commented:
@kelson42 I updated the PR (rolled back the changes that went beyond just fixing #1184 and made the CI pass). If the PR is going to be merged that way its commits have to be squashed into one.

kelson42 commented:
@veloman-yunkan Thank you for the comment
@aditii2712 Would you be able to fix the few last details please so we can merge your code?
